ginga/util/catalog.py

```diff
@@ -229,6 +235,169 @@ def _search(self, center, radius, catalog):
+
+        v = Vizier(columns=columns, column_filters=column_filters)
+        results = Vizier.query_region(center, radius, catalog=catalog)
+
+        return results[0]
+
+    def _search_box(self, center, width, height, catalog, columns, column_filters):
+        # override this method to pass some special kwargs to the search
+        if columns:
+            columns.insert(0, self.mapping['id'])
+            columns.insert(1, self.mapping['ra'])
+            columns.insert(2, self.mapping['dec'])
+
+        v = Vizier(columns=columns, column_filters=column_filters)
+        v.ROW_LIMIT = 500
+        results = v.query_region(center, width=width, height=height, catalog=catalog)
+
+        return results[0]
+
+    def search(self, **params):
+        """
+        For compatibility with generic star catalog search.
+        """
+
+        self.logger.debug("search params=%s" % (str(params)))
+        ra, dec = params['ra'], params['dec']
+        if not (':' in ra):
+            # Assume RA and DEC are in degrees
+            ra_deg = float(ra)
+            dec_deg = float(dec)
+        else:
+            # Assume RA and DEC are in standard string notation
+            ra_deg = wcs.hmsStrToDeg(ra)
+            dec_deg = wcs.dmsStrToDeg(dec)
+
+        # Note requires astropy 0.3.x+
+        c = coordinates.SkyCoord(ra_deg * units.degree,
+                                 dec_deg * units.degree,
+                                 frame='icrs')
+
+        self.logger.info("Querying catalog: %s" % (self.full_name))
+        time_start = time.time()
+        with warnings.catch_warnings():  # Ignore VO warnings
+            warnings.simplefilter('ignore')
+            if params['radius'] == 1:
+                # Convert to degrees for search radius
+                radius_deg = float(params['r']) / 60.0
+                # radius_deg = float(params['r'])
+                results = self._search_radius(c, radius_deg * units.degree,
+                                              self.full_name, columns=self.cat_columns, column_filters=self.cat_column_filters)
+            elif params['box'] == 1:
+                # Convert to degrees for search width
+                width_deg = float(params['width']) / 60.0
+                # Convert to degrees for search height
+                height_deg = float(params['height']) / 60.0
+                results = self._search_box(c, width_deg * units.degree, height_deg * units.degree,
+                                           self.full_name, columns=self.cat_columns, column_filters=self.cat_column_filters)
+            else:
+                #attempt to use the radius, this shouldn't happen
+                # Convert to degrees for search radius
+                radius_deg = float(params['r']) / 60.0
+                # radius_deg = float(params['r'])
+                results = self._search_radius(c, radius_deg * units.degree,
+                                              self.full_name)
+
```

Comment:
and here to catch the exception:
```python
except Exception as e:
    self.logger.error(f"vizier query raised an exception: {e}", exc_info=True)
    results = None
```
Just in case there was some unusual exception doing the query.